### PR TITLE
[Concept Entry] Postgresql window-functions #4591

### PR DESCRIPTION
**Pull request by Johnnie-kay**
### Description

- Created a new concept entry for Postgresql window functions.
- Closes issue #4591 

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/postgresql/concepts/window-functions/window-functions.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+| 01 | John     | Tokyo   | +81     | 5500   |
+| 02 | Kate     | Beijing | +186    | 5000   |
+| 03 | Tad      | Mumbai  | +91     | 2500   |
+| 04 | Martha   | Tokyo   | +727    | 3500   |
+| 05 | Will     | paris   | +33     | 4000   |
+| 06 | Ruth     | Dubai   | +971    | 2500   |
+| 07 | Kaka     | Mumbai  | +213    | 4500   |
+| 08 | Tolf     | Tokyo   | +44     | 2000   |
+
+From the above _staff_ table, to rank the employees using the `Rank()` function, we select the id, salary, and address columns from the _staff_ table, and then use the `RANK()` window function to assign a rank to each row based on the _salary_ column within each partition defined by the _address_ column.
```
Comment (SaviDahegaonkar) [inline content/postgresql/concepts/window-functions/window-functions.md:52]:
```suggestion
From the above `_staff_ table`, to rank the employees using the `Rank()` function, we select the `id`, `salary`, and `address` columns from the `_staff_ ` table, and then use the `RANK()` window function to assign a rank to each row based on the ` _salary_ ` column within each partition defined by the `_address_ ` column.
```

File: content/postgresql/concepts/window-functions/window-functions.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+#Syntax:
+
+`SELECT id, salary, address, Rank() OVER (PARTITION BY address ORDER BY salary DESC) FROM staff;`
```
Comment (SaviDahegaonkar) [inline content/postgresql/concepts/window-functions/window-functions.md:56]:
```suggestion

SELECT id, salary, address, Rank() OVER (PARTITION BY address ORDER BY salary DESC) FROM staff;
```
This is not a syntax as we are using it to print something. Write the code in the code block (```sql). Check the [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) for the reference.

File: content/postgresql/concepts/window-functions/window-functions.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+Breaking down the syntax:
+- The clause `SELECT`, selects the id, salary, and address colums from the staff table.
+- The `Rank() OVER` part computes the row number of each row.
+- The `(PARTITION BY address ORDER BY salary DESC)` clause actively defines how the window function partitions the data.
+- The statement ends with `FROM staff`, signifying that it operates on the staff table.
```
Comment (SaviDahegaonkar) [inline content/postgresql/concepts/window-functions/window-functions.md:62]:
```suggestion
Breaking down the syntax:
- The clause `SELECT`, selects the id, salary, and address columns from the staff table.
- The `Rank() OVER` part computes the row number of each row.
- The `(PARTITION BY address ORDER BY salary DESC)` clause actively defines how the window function partitions the data.
- The statement ends with `FROM staff`, signifying that it operates on the `_staff _` table.
```

File: content/postgresql/concepts/window-functions/window-functions.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+| id | salary | adrress | rank | 
+|--- | -----  | -----   | ---- |
+| 01 |  5500  | Tokyo   |  1   |
+| 04 |  3500  | Tokyo   |  2   |
+| 08 |  2000  | Tokyo   |  3   |
+| 07 |  4500  | Mumbai  |  1   |
+| 03 |  2500  | Mumbai  |  2   |
+| 02 |  5000  | Beijing |  1   |
+| 05 |  4000  | Paris   |  1   |
+| 06 |  2500  | Dubai   |  1   |
```
Comment (SaviDahegaonkar) [inline content/postgresql/concepts/window-functions/window-functions.md:75]:
```suggestion
## Output

| `id` | `salary`| `address` | `rank` | 
|--- | -----  | -----   | ---- |
| 01 |  5500  | Tokyo   |  1   |
| 04 |  3500  | Tokyo   |  2 |
| 08 |  2000  | Tokyo   |  3 |
| 07 |  4500  | Mumbai | 1 |
| 03 |  2500  | Mumbai | 2 |
| 02 |  5000  | Beijing | 1 |
| 05 |  4000  | Paris    | 1 |
| 06 |  2500  | Dubai | 1 |
```
Add a line like this after the subheading `## Output`: 
`This is the output of the above query:`

File: content/postgresql/concepts/window-functions/window-functions.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+
+In PostgreSQL, the `row_number` window function plays a useful role by defining the row number of a column. Let's use the _staff_ table to assign a row number to each row based on the `salary` column, within each group of rows that have the same `address`, ordering the rows by `salary` in descending order within each group.
```
Comment (SaviDahegaonkar) [inline content/postgresql/concepts/window-functions/window-functions.md:78]:
```suggestion
In PostgreSQL, the `row_number` window function plays a useful role by defining the row number of a column. Let's use the `_staff_ table` to assign a row number to each row based on the `salary` column, within each group of rows that have the same `address`, ordering the rows by `salary` in descending order within each group.
```
Just added backticks.

File: content/postgresql/concepts/window-functions/window-functions.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+#Syntax:
+
+`SELECT id, salary, address, row_number() OVER (PARTITION BY address ORDER BY salary DESC) FROM staff`
```
Comment (SaviDahegaonkar) [inline content/postgresql/concepts/window-functions/window-functions.md:82]:
```suggestion

SELECT id, salary, address, row_number() OVER (PARTITION BY address ORDER BY salary DESC) FROM staff
```
Please implement the changes here as I told you in the previous comment.

File: content/postgresql/concepts/window-functions/window-functions.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+Breaking down the syntax:
+- The clause `SELECT`, selects the id, salary, and address colums from the staff table.
+- The `Row_number() OVER` part computes the row number of each row.
+- The `(PARTITION BY address ORDER BY salary DESC)` clause actively defines how the window function partitions the data.
+- The statement ends with `FROM staff`, signifying that it operates on the staff table.
```
Comment (SaviDahegaonkar) [inline content/postgresql/concepts/window-functions/window-functions.md:88]:
``` suggestion
Breaking down the syntax:
- The clause `SELECT`, selects the `id`, `salary`, and `address` columns from the `_staff_` table.
- The `Row_number() OVER` part computes the row number of each row.
- The `(PARTITION BY address ORDER BY salary DESC)` clause actively defines how the window function partitions the data.
 - `PARTITION BY address` partitions the rows into groups based on values in the `address` column.
 - `ORDER BY salary DESC` allows the rows to be ordered by `salary` in descending order within each partition.
- The statement ends with `FROM staff`, signifying that it operates on the `_staff_` table.
```

File: content/postgresql/concepts/window-functions/window-functions.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+| id | salary | adrress | row_number | 
+|--- | -----  | -----   | ---- |
+| 01 |  5500  | Tokyo   |  1   |
+| 04 |  3500  | Tokyo   |  2   |
+| 08 |  2000  | Tokyo   |  3   |
+| 07 |  4500  | Mumbai  |  1   |
+| 03 |  2500  | Mumbai  |  2   |
+| 02 |  5000  | Beijing |  1   |
+| 05 |  4000  | Paris   |  1   |
+| 06 |  2500  | Dubai   |  1   |
```
Comment (SaviDahegaonkar) [inline content/postgresql/concepts/window-functions/window-functions.md:101]:
```suggestion
## Output

| `id` | `salary` | `address` | `row_number` | 
|--- | -----  | -----   | ---- |
| 01 | 5500 | Tokyo |  1  |
| 04 | 3500 | Tokyo   |  2   |
| 08 | 2000 | Tokyo   |  3   |
| 07 | 4500 | Mumbai | 1  |
| 03 | 2500 | Mumbai | 2 |
| 02 |  5000  | Beijing | 1 |
| 05 |  4000  | Paris   | 1 |
| 06 |  2500  | Dubai | 1 |
```
Write a line saying `This is the output of the above query:` before the output.

**Review by SaviDahegaonkar — CHANGES_REQUESTED** [review]
Hello Author,  I have suggested a few changes in the file and corrected some minor mistakes accordingly. Please have a look at it and please let me know.
Thanks!

**Comment by Johnnie-kay** [comment]
Hello @SaviDahegaonkar, all corrections and suggestions have been attended to.

**Comment by SaviDahegaonkar** [comment]
> Hello @SaviDahegaonkar, all corrections and suggestions have been attended to. 

Yes I will look into it.
Thank you.

**Review by Johnnie-kay — COMMENTED** [review]
I have reviewed the window-functions.md file and made all corrections

**Review by mohitO12-dev — APPROVED** [review]

**Review by SaviDahegaonkar — APPROVED** [review]

**Pull request merged**